cmd/proxy-server/main.go

```diff
@@ -132,10 +132,17 @@ func runServer(cCtx *cli.Context) error {
 		return errors.New("not all of --tls-certificate-path and --tls-private-key-path specified")
 	}
 
-	serverAttestationType, err := proxy.ParseAttestationType(serverAttestationTypeFlag)
+	// Auto-detect server attestation type if not specified
+	serverAttestationType, err := proxy.ParseAttestationType(cCtx.String("server-attestation-type"))
 	if err != nil {
-		log.With("attestation-type", cCtx.String("server-attestation-type")).Error("invalid server-attestation-type passed, see --help")
-		return err
+		// If parsing fails and no type was specified, use auto-detection
+		if cCtx.String("server-attestation-type") == "" {
+			serverAttestationType = proxy.DetectAttestationType()
+			log.With("detected_attestation", serverAttestationType).Info("Auto-detected server attestation type")
+		} else {
+			log.With("attestation-type", cCtx.String("server-attestation-type")).Error("invalid server-attestation-type passed, see --help")
+			return err
+		}
 	}
 
 	clientAttestationType, err := proxy.ParseAttestationType(cCtx.String("client-attestation-type"))
```

Comment:
can `client-attestation-type` also be detected? ideally we'd run a single server proxy instance that works for all client attestation types

Reply:
Yes, I've added this feat. Client attestation type is now determined via the measurements.json. During TLS handshake attestation the attester will check the OID of the attestee and then select the measurements it has in store for this OID, and tries to verify if they are a match. This works vice versa for the proxy-client, it attests servers based on measurement attestation types as well.